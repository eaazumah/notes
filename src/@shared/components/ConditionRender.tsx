import React from 'react';

interface RenderProps {
  renderIf: boolean;
  children?: React.ReactNode;
}
export type ConditionRenderProps<T = Record<string, any>> = T & RenderProps;

const ConditionRender = <P extends object>(
  Component: React.ComponentType<P>,
) => {
  return function ({renderIf, ...props}: ConditionRenderProps<P>) {
    if (!renderIf) return null;
    return <Component {...(props as P)} />;
  };
};

export const ConditionRenderComponent: React.FC<RenderProps> = ({
  renderIf,
  children,
}) => {
  if (!renderIf) return null;
  return <>{children}</>;
};

export default ConditionRender;
