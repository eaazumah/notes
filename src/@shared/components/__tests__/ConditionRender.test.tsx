import React from 'react';
import {render} from '@testing-library/react-native';
import ConditionRender, {ConditionRenderComponent} from '../ConditionRender';
import {Text} from 'react-native';

const TestComponent: React.FC = () => {
  return <Text>Some content to be rendered</Text>;
};

describe('ConditionRender', () => {
  it('should render component', () => {
    const Component = ConditionRender(TestComponent);
    const screen = render(<Component renderIf />);

    expect(screen.toJSON()).toMatchSnapshot();
    expect(screen.getByText(/Some content to be rendered/)).toBeDefined();
  });

  it('should not render component', () => {
    const Component = ConditionRender(TestComponent);
    const screen = render(<Component renderIf={false} />);

    expect(screen.toJSON()).toMatchSnapshot();
    expect(screen.queryByText(/Some content to be rendered/)).toBeNull();
  });
});

describe('ConditionRenderComponent', () => {
  it('should render component', () => {
    const screen = render(
      <ConditionRenderComponent renderIf>
        <TestComponent />
      </ConditionRenderComponent>,
    );

    expect(screen.toJSON()).toMatchSnapshot();
    expect(screen.getByText(/Some content to be rendered/)).toBeDefined();
  });

  it('should not render component', () => {
    const screen = render(
      <ConditionRenderComponent renderIf={false}>
        <TestComponent />
      </ConditionRenderComponent>,
    );

    expect(screen.toJSON()).toMatchSnapshot();
    expect(screen.queryByText(/Some content to be rendered/)).toBeNull();
  });
});
