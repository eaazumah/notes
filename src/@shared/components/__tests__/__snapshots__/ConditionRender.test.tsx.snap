// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionRender should not render component 1`] = `null`;

exports[`ConditionRender should render component 1`] = `
<Text>
  Some content to be rendered
</Text>
`;

exports[`ConditionRenderComponent should not render component 1`] = `null`;

exports[`ConditionRenderComponent should render component 1`] = `
<Text>
  Some content to be rendered
</Text>
`;
